### ISSUE TITLE
Questions re the (upcoming?) 2.x release and its impact on third party integrations

### ISSUE THREAD

**Issue by indigoxela** [Leaflet/Leaflet#9238]
### Checklist

- [ ] I've searched through the [plugins](https://leafletjs.com/plugins.html) to make sure this feature isn't already available, or think it shouldn't require a plugin.
- [X] I've searched through the [current issues](https://github.com/Leaflet/Leaflet/issues) to make sure this feature hasn't been requested already.
- [X] I agree to follow the [Code of Conduct](https://github.com/Leaflet/Leaflet/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.

### Motivation

Many thanks to all the contributors working on the next release, which will probably be 2.x. :heart: 

However, several times I read in this queue's issues about breaking changes, and as an integration project maintainer, this makes me concerned, that my stuff might not be compatible.

So my questions:

- What are steps I'd have to take to make sure, my project will be compatible with 2.x? More specific: If my current setup works without any deprecation nagging in the console, will it be compatible automatically?
- Is there already some documentation re the breaking changes (detailed or not)?
- When will the 2.x release be ready? Any date or time window scheduled, yet?
- When 2.x will get released, what's the time window until 1.x will get deprecated?

Sorry, didn't find any better place or issue category for my questions. :wink: I'd bet, I'm not the only one being concerned.

### Suggested solution

I'm not sure if it's not too early, but opening a meta issue for 2.x release in this queue, which lists possible problems for integrations would be a huge help.

In other words: inform about things to expect.

### Alternatives considered

I could just wait until 2.x is released and figure out what I need to do then (by trial and error).

But as there are already 411 commits since v1.9.4 as of today, this could turn out to be a tricky task.

**Comment by Falke-Design** [comment]
Hi @indigoxela,

we have already done some changes for v2 but we are still far away from a release. But I will answer your questions one by one:

> What are steps I'd have to take to make sure, my project will be compatible with 2.x? More specific: If my current setup works without any deprecation nagging in the console, will it be compatible automatically?

It depends on how much you have used the `DOMUtil` and `Util` functions. We removed many of them, because there are now easy to use vanilla JavaScript functions for that. But in generall you should have not much troubles.

> Is there already some documentation re the breaking changes (detailed or not)?

We currently don't have a documentation but you can look into the `v2` milestone: https://github.com/Leaflet/Leaflet/milestone/39 
The most important changes are:
* dropping IE support
* Switching to ESM
* PointerEvents

> When will the 2.x release be ready? Any date or time window scheduled, yet?

We are currently not able to give you a date but it looks like we will not release in the next 6 months. Maybe with the end of the year.

> When 2.x will get released, what's the time window until 1.x will get deprecated?

`1.9` was already the last minor version release. We will only provide new releases for important bugs and without any new features. I can't say if we will continue fixing bugs in `1.9` after releasing `v2`

**Comment by indigoxela** [comment]
@Falke-Design many thanks for taking the time for this thorough response. :pray: 

The L.DomEvent part is the one that concerns me the most, as it's used in actually all plugins, my integration modules for a web CMS use. :grimacing: And I'm not sure about the maintenance status of some of those plugins.

>  it looks like we will not release in the next 6 months. Maybe with the end of the year.

Many thanks for this clarification. I understand, that providing a definitive date just isn't possible, yet. Narrowing the time frame already helps.

> 1.9 was already the last minor version release. We will only provide new releases for important bugs and without any new features. I can't say if we will continue fixing bugs in 1.9 after releasing v2

Understandable, that fully maintaining two branches in parallel isn't feasible. I'm relieved, that you consider fixing outstanding bugs (like security).

Looks like I won't be an early bird with switching to 2.x because of plugins :wink: , but your answers help me a lot with planning.

And again, many thanks for all your hard work on this cool project. :heart:

**Comment by Falke-Design** [comment]
Your welcome 😄 

> The L.DomEvent part is the one that concerns me the most

should be not really change, only `DomUtil` and the `Utils` functions.
